cgp/genome.py

```diff
@@ -354,79 +357,82 @@ def mutate(self, mutation_rate: float, rng: np.random.RandomState):
         mutation_rate : float
             Proportion of genes to be mutated, between 0 and 1.
         rng : numpy.random.RandomState
-            Random number generator instance to use for crossover.
+            Random number generator instance to use for mutation.
 
         Returns
         ----------
         bool
             True if only inactive regions of the genome were mutated, False otherwise.
         """
 
-        def count_dna_differences(dna0: List[int], dna1: List[int]) -> int:
-            return len([1 for gene0, gene1 in zip(dna0, dna1) if gene0 != gene1])
-
-        n_mutations = int(mutation_rate * len(self.dna))
-        assert n_mutations > 0
-
         graph = CartesianGraph(self)
         active_regions = graph.determine_active_regions()
-
         dna = list(self._dna)
-
         only_silent_mutations = True
-        while count_dna_differences(self.dna, dna) < n_mutations:
 
-            gene_idx = rng.randint(0, self._n_genes)
-            region_idx = gene_idx // self._length_per_region
+        selected_gene_indices = np.nonzero(np.random.rand(len(dna)) < mutation_rate)[0]
 
-            if self._is_input_region(region_idx):
-                continue  # nothing to do here
+        for (allele_idx, allele) in zip(
```

Comment:
sorry to be annoying, but i think `gene_idx` is fine, this refers to a particular position in the genome. really it should be called just `gene` but i think it's ok to add the suffix. `allele_idx` could be misinterpreted as an index in the list of possible settings for this particular gene.

Reply:
Ah yes makes sense, will change back to  `gene_idx`